.github/workflows/build.yml

```diff
@@ -0,0 +1,85 @@
+name: Build
+
+on:
+  push:
+    branches:
+      - 'main'
+  pull_request:
+
+jobs:
+  build:
+    name: Build
+    runs-on: windows-latest
+    steps:       
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
+      # Environment things
+      - name: Add Msbuild To PATH
+        uses: microsoft/setup-msbuild@v1.0.2
+        with:
+         vs-version: '16.5'
+
+      - name: Setup Boost Environment
+        run: |
+          echo "BOOST_ROOT=${env:BOOST_ROOT_1_72_0}\\include" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+          echo "BOOST_LIBRARYDIR=${env:BOOST_ROOT_1_72_0}\\lib" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+
+      - name: Cache OpenSSL
+        id: cache-openssl
+        uses: actions/cache@v2.1.3
+        with:
+          path: "C:\\Program Files (x86)\\OpenSSL-Win32"
+          key: "ukcp-openssl"
+
+      - name: Install OpenSSL
+        if: steps.cache-openssl.outputs.cache-hit != 'true'
+        run: choco install openssl --forcex86 | out-null
+
+      - name: Setup OpenSSL Environment
+        run: |
+          echo "OPENSSL_ROOT=C:\\Program Files (x86)\\OpenSSL-Win32\\include" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+          echo "OPENSSL_LIBRARYDIR=C:\\Program Files (x86)\\OpenSSL-Win32\\lib\\VC\\static" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+
+      - name: Restore Nuget Packages
+        run: |
+          cd msvc
+          nuget restore
+          cd ..
+
+      # If we're running on main, get the next release version
+      - name: Setup Build Version
+        if: github.ref == 'refs/heads/main'
+        run: |
+          $semanticRelease = npx semantic-release@15 --dry-run | Select-String -Pattern "Published release ((0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)"

```

Comment:
Holy regex batman!

Reply:
Yep - semantic isn’t too great for these sorts of builds but it works :D